Shop/MainWindow.xaml.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace Shop
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        private ObservableCollection<Product> myList = new ObservableCollection<Product>();
+
+        public MainWindow()
+        {
+            InitializeComponent();
+
+            this.SetDefaultTextFields();
+            this.pCodeTextBox.Focus();
+
+            this.myDataGrid.ItemsSource = myList;
+        }
+
+        //
+        // Purification of the input fields. Setting the focus to the input field.
+        //
+        private void SetDefaultTextFields()
+        {
+            this.pCodeTextBox.Text = string.Empty;
+            this.productNameTextBox.Text = string.Empty;
+            this.quantityTextBox.Text = string.Empty;
+            this.priceTextBox.Text = string.Empty;
+
+            this.pCodeTextBox.Focus();
+        }
+
+        //
+        // Switch background color when entering illegal characters.
+        //
+        private void ChangeTextBoxBackgroundColor(string textBoxName, SolidColorBrush backgroundErrorColor)
+        {
+            switch (textBoxName)
+            {
+                case "pCodeTextBox":
+                    this.pCodeTextBox.Background = backgroundErrorColor;
```

Comment:
All these cases manages the same color. Perhaps it would be easier to receive appropriate textbox and in the end of method once set property Background?
Also do not use literals. Move to constants or enum.